### PR TITLE
Add module global type information APIs

### PR DESCRIPTION
**Pull request by bnason-nf**

### CODE REVIEW

File: core/iwasm/interpreter/wasm.h [hunk 1]
```diff
@@ -584,11 +584,15 @@ typedef struct WASMTagImport {
 } WASMTagImport;
 #endif
 
+typedef struct WASMGlobalCommon {
+    uint8 type;
+    bool is_mutable;
+} WASMGlobalCommon;
+
```
Comment (lum1n0us) [inline core/iwasm/interpreter/wasm.h:591]: just my personal suggestion. 

```c
typedef struct WASMGlobalType {
    uint8 val_type;
    bool  is_mutable;
} WASMGlobalType;
```

File: core/iwasm/interpreter/wasm.h [hunk 2]
```diff
@@ -584,11 +584,15 @@ typedef struct WASMTagImport {
...
 typedef struct WASMGlobalImport {
     char *module_name;
     char *field_name;
-    uint8 type;
-    bool is_mutable;
+    WASMGlobalCommon common;
```
Comment (lum1n0us) [inline core/iwasm/interpreter/wasm.h:595]:
```suggestion
    WASMGlobalCommon type;
```

less confusion.

File: core/iwasm/interpreter/wasm.h [hunk 3]
```diff
@@ -705,8 +709,7 @@ struct WASMTag {
 #endif
 
 struct WASMGlobal {
-    uint8 type;
-    bool is_mutable;
+    WASMGlobalCommon common;
```
Comment (lum1n0us) [inline core/iwasm/interpreter/wasm.h:712]:
```suggestion
    WASMGlobalCommon type;
```

less confusion

**Comment by lum1n0us** [comment]
Plus, I kindly thought we should use `wasm_runtime_get_import_type` and `wasm_runtime_get_export_type` to implement `wasm_module_exports` and `wasm_module_imports` in order to keep the code simple.

File: core/iwasm/interpreter/wasm.h (hunk 1 above)
Comment (bnason-nf) [reply core/iwasm/interpreter/wasm.h:591]: I'll make this change.

File: core/iwasm/interpreter/wasm.h (hunk 2 above)
Comment (bnason-nf) [reply core/iwasm/interpreter/wasm.h:595]: I'll make this change.

File: core/iwasm/interpreter/wasm.h (hunk 3 above)
Comment (bnason-nf) [reply core/iwasm/interpreter/wasm.h:712]: I'll make this change.

**Comment by bnason-nf** [comment]
> Plus, I kindly thought we should use `wasm_runtime_get_import_type` and `wasm_runtime_get_export_type` to implement `wasm_module_exports` and `wasm_module_imports` in order to keep the code simple.

I think we can do that once all the import/export types exposed in wasm_export.h. After this change lands and globals are exposed, then also memory and tables would need to be exposed. Once functions, globals, memory, and tables are all exposed, then some of the internals of `wasm_module_exports` and `wasm_module_exports` can be converted to use those new APIs. Unless maybe I'm missing something, and you see a way to partially convert those function internals?

**Review by lum1n0us — APPROVED** [review]
LGTM

**Review by wenyongh — APPROVED** [review]

**Pull request merged**